docs/contracts/v4/guides/15-single-sided-liquidity.mdx

```diff
@@ -0,0 +1,423 @@
+
+PoolKey memory pool = PoolKey({
+    currency0: currency0, // imported from Config.sol
+    currency1: currency1, // imported from Config.sol
+    fee: LP_FEE,
+    tickSpacing: TICK_SPACING,
+    hooks: hookContract
+});
+
+// Get current pool sqrtPrice and tick
+(uint160 sqrtPriceX96,,,) = POOLMANAGER.getSlot0(pool.toId()); //POOLMANAGER is imported from Constants.sol
+int24 currentTick = TickMath.getTickAtSqrtPrice(sqrtPriceX96);
+```
+
+Next we decrease the `Price` by 5%. We create a new Price with a numerator 5% lower than our current Price:
+
+```solidity
+uint160 targetSqrtPriceX96 = uint160((uint256(sqrtPriceX96) * 974679) / 1e6); //Since we need to calculate 95% of the current sqrtprice
+```
+
+We have calculated our target Price but we still need to find the **nearest usable tick** to create our Position.
+
+:::info
+As Positions can only start and end at initializable Ticks of the Pool, so we can only create a Range Order to a Price that exactly matches an initializable Tick.
+:::
+
+We make use of the getTickAtSqrtPrice of TickMath contract to calculate the tick at a specific sqrtPrice value.
+
+```solidity
+import {TickMath} from "v4-core/src/libraries/TickMath.sol";
+
+int24 targetTick = TickMath.getTickAtSqrtPrice(targetSqrtPriceX96);
+int24 tickUpper = (targetTick / TICK_SPACING) * TICK_SPACING;
+```
+
+This nearest Tick will most likely not **exactly** match our Price target.
+
+Depending on our personal preferences we can either err on the higher or lower side of our target by adding or subtracting the `tickSpacing` if the initializable Tick is lower or higher than the theoretically closest Tick.
+
+### Lower Tick
+
+We now find the lower Tick by subtracting the tickSpacing from the upper Tick:
+
+```solidity
+int24 tickLower = tickUpper - TICK_SPACING;
+```
+
+If the price difference is too low, the lower tick may be left of the current Tick of the Pool.
+In that case we would not be able to provide single side liquidity.
+We can either throw an Error or increase our Position by one Tick.
+
+```solidity
+if (tickLower <= currentTick) {
+    tickLower += TICK_SPACING;
+    tickUpper += TICK_SPACING;
+}
+```
+
+We now have a lower and upper Tick for our Position, next we need to construct and mint it.
+
+## Creating the Single Side Liquidity Position
+
+We will first calculate the liquidity corresponding to our token1, tickLower and tickUpper.
+
+```solidity
+uint128 liquidity = LiquidityAmounts.getLiquidityForAmount1(
+    TickMath.getSqrtPriceAtTick(tickLower),
+    TickMath.getSqrtPriceAtTick(tickUpper),
+    token1Amount
+);
+```
+
+And then, we recalculate actual token1 necessary for that liquidity as we round up the liqudity to the nearest number and that could have changed the required token1.
+
+```solidity
+uint256 requiredToken1 = LiquidityAmounts.getAmount1ForLiquidity(
+    TickMath.getSqrtPriceAtTick(tickLower),
+    TickMath.getSqrtPriceAtTick(tickUpper),
+    liquidity
```

Comment:
lets trim this section and mention that we can use LiquidityAmounts library (with link) to calculate the token amounts needed